src/ralph/virtual/tests/test_admin.py

```diff
@@ -0,0 +1,86 @@
+# -*- coding: utf-8 -*-
+import json
+
+from django.core.exceptions import ValidationError
+
+from ralph.admin.sites import ralph_site
+from ralph.assets.models.base import BaseObject
+from ralph.assets.tests.factories import BaseObjectFactory
+from ralph.data_center.models import DataCenterAsset
+from ralph.data_center.tests.factories import (
+    DataCenterAssetFactory,
+    ServiceEnvironmentFactory
+)
+from ralph.licences.tests.factories import LicenceFactory
+from ralph.tests import RalphTestCase
+from ralph.virtual.admin import VirtualServerAdmin, VirtualServerForm
+from ralph.virtual.models import VirtualServer
+from ralph.virtual.tests.factories import CloudHostFactory, VirtualServerFactory
+
+
+class MockAdmin:
+    def has_perm(self, perm):
+        return True
+
+
+class MockRequest:
+    pass
+
+
+class TestVirtualServerForm(RalphTestCase):
+    
+    def setUp(self):
```

Comment:
check out these places for inspiration how to make requests to admin in tests:
https://github.com/allegro/ralph/blob/ng/src/ralph/data_center/tests/test_admin.py#L29-L34
https://github.com/allegro/ralph/blob/ng/src/ralph/data_center/tests/test_admin.py#L69-L77

Reply:
Done.